shm/Map.cpp

```diff
@@ -1,6 +1,30 @@
 #include "Map.hpp"
-
 #include <algorithm>
+#include <random>
+#include <iostream>
+
+Map::Map(){
+    islandsLocations_.reserve(amountIslands);
+    generateIslands();
+}
+
+void Map::generateIslands() {
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> distrib(widthIsland, heightIsland);
+    
+      for (size_t i = 0; i < amountIslands; i++) { 
+        size_t positionX; 
+        size_t positionY; 
+        do { 
+            size_t positionX = distrib(gen); 
+            size_t positionY = distrib(gen);
+        } while (std::any_of(islandsLocations_.begin(), islandsLocations_.end(), [positionX, positionY](auto& island) { 
+            return island.getPosition() == Coordinates(positionX, positionY); 
+        })); 
+        islandsLocations_.push_back(Island(positionX, positionY)); 
```

Comment:
I'm not sure if we can use `emplace_back` instead of `push_back`. Maybe someone could tell more about it? @PierwszyKrytykRP what do you think?

Reply:
It can be used since it will replace Island(x,y) notation

Reply:
Alternatively instead of directly pushing you could use existing function Map::addIsland like so `addIsland({positionX, positionY})`